supporting-blog-content/agentic-rag/agent_rag_news_assistant.ipynb

```diff
@@ -0,0 +1,807 @@
+     "text": [
+      "\u001B[32m2025-10-27 12:21:45.407\u001B[0m | \u001B[1mINFO    \u001B[0m | \u001B[36m__main__\u001B[0m:\u001B[36mself_reflection\u001B[0m:\u001B[36m29\u001B[0m - \u001B[1mSelf-reflection passed — binary_score=True\u001B[0m\n",
+      "\u001B[32m2025-10-27 12:21:46.972\u001B[0m | \u001B[1mINFO    \u001B[0m | \u001B[36m__main__\u001B[0m:\u001B[36m<module>\u001B[0m:\u001B[36m2\u001B[0m - \u001B[1m\n",
+      "Final Summary:\n",
+      ": In June 2024, several notable AI models were launched or published, including Runway's Gen-3 Alpha, Anthropic's Claude 4, Google's Veo 3 (which generates synchronized audio and video), Mistral's open-source Devstral model, and Alibaba's Qwen3 series.\u001B[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "result = agent.invoke({\"query\": query1})\n",
+    "logger.info(f\"\\nFinal Summary:\\n: {result['summary']}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "id": "a63de923-ca7b-42fb-9a80-7f261cb92d74",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\u001B[32m2025-10-27 12:22:25.073\u001B[0m | \u001B[1mINFO    \u001B[0m | \u001B[36m__main__\u001B[0m:\u001B[36mrouter\u001B[0m:\u001B[36m11\u001B[0m - \u001B[1mRouter selected the datasource: vectorstore\u001B[0m\n",
+      "\u001B[32m2025-10-27 12:22:25.075\u001B[0m | \u001B[1mINFO    \u001B[0m | \u001B[36m__main__\u001B[0m:\u001B[36mrouter\u001B[0m:\u001B[36m12\u001B[0m - \u001B[1mUser query: What technological innovations are discussed in Sci/Tech news?\u001B[0m\n",
+      "/Users/kirtisodhi/agentic-rag/path/to/venv/lib/python3.13/site-packages/langchain_elasticsearch/_sync/vectorstores.py:530: ElasticsearchWarning: text_expansion is deprecated. Use sparse_vector instead.\n",
+      "  hits = self._store.search(\n",
+      "\u001B[32m2025-10-27 12:22:25.998\u001B[0m | \u001B[1mINFO    \u001B[0m | \u001B[36m__main__\u001B[0m:\u001B[36mself_reflection\u001B[0m:\u001B[36m29\u001B[0m - \u001B[1mSelf-reflection passed — binary_score=True\u001B[0m\n",
+      "\u001B[32m2025-10-27 12:22:27.416\u001B[0m | \u001B[1mINFO    \u001B[0m | \u001B[36m__main__\u001B[0m:\u001B[36m<module>\u001B[0m:\u001B[36m2\u001B[0m - \u001B[1m\n",
+      "Final Summary:\n",
+      ": Recent Sci/Tech news highlights several technological innovations: NASA is collaborating with Silicon Valley firms to build a powerful Linux-based supercomputer for theoretical research and shuttle engineering; Genetic Savings & Clone has developed chromatin transfer technology to clone cats; Princeton scientists report that existing technologies could immediately help stabilize global warming; and a set of micro-games for GameBoy was recognized for its innovation at a gaming festival. Additionally, cybersecurity advances are featured in The Washington Post's special report.\u001B[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "result = agent.invoke({\"query\": query2})\n",
```

Comment:
This will not run for me as `elastic_vectorstore` is undefined due to the timeout above.

Reply:
Adding a timeout options solves this